add FAQ entry, remove custom branch from PR template page (#2679)
fishtown-analytics/docs.getdbt.com: website/docs/faqs/Environments/custom-branch-settings.md

---
title: How do I use the `Custom Branch` settings in a dbt Cloud Environment?
description: "Use custom code from your repository"
sidebar_label: 'Custom Branch settings'
id: custom-branch-settings
---

> dbt Cloud environments allow users to specify custom git settings to use different versions of their dbt project repositories other than their default branch. After checking off the `Only run on a custom branch` checkbox in the `General Settings` section of the environment settings page, users can specify the particular **branch name or tag** in the`Custom Branch` setting that dbt will use when executing in that environment. Depending on the type of environment, this setting will have slightly different effects.

Reviewer: ```suggestion

In dbt Cloud environments, you can change your git settings to use a different branch in your dbt project repositories besides the default branch. When you make this change, you run dbt on a custom branch. When specified, dbt Cloud executes models using the custom branch setting for that environment. Development and deployment environments have slightly different effects.

To specify a custom branch:
1. Edit an existing environment or create a new one
2. Select **Only run on a custom branch** under General Settings
3. Specify the **branch name or tag**

```

> ## In development

Reviewer: ```suggestion
## Development
```

> In a development environment, dbt Cloud will reference the default branch in connected repositories (most commonly the `main` branch) in the IDE as the read only brnach that developers will create development branches from. The `Custom Branch` setting overrides that behavior if there is another branch that should be used. Users will not be able to make commits to the branch specified by this setting.  

Reviewer: ```suggestion
In a development environment, the default branch (commonly the `main` branch) is a read-only branch found in the IDE's connected repositories, which you can use to create development branches. Identifying a custom branch overrides this default behavior. Instead, your custom branch becomes read-only and can be used to create development branches. You will no longer be able to make commits to the custom branch. 
```

> For example, you can use the `develop` branch of a connected repository. Edit an environment, then in "General settings" select  **Only run on a custom branch** , and in "Custom branch" type **develop** or the name of your custom branch.

Reviewer: ```suggestion
For example, you can use the `develop` branch of a connected repository. Edit an environment, select  **Only run on a custom branch** in **General settings** , enter **develop** as the name of your custom branch.
```

<Lightbox src="/img/docs/dbt-cloud/cloud-configuring-dbt-cloud/dev-environment-custom-branch.png" title="Configuring a custom base repository branch"/>

> ## In deployment

Reviewer: ```suggestion
## Deployment
```

> When running jobs in a deployment environment, dbt will clone your project from your connected repository before executing your models. By default, dbt will pull from the default branch of your repository (most commonly the `main` branch). To control which version of your project is executed during job runs for a particular environment you can edit the `Custom Branch` setting as above to specify a different version of the project. Accepted values for this setting are **a branch name or a tag name.**

Reviewer: ```suggestion
When running jobs in a deployment environment, dbt will clone your project from your connected repository before executing your models. By default, dbt uses the default branch of your repository (commonly the `main` branch). To specify a different version of your project for dbt to execute during job runs in a particular environment, you can edit the Custom Branch setting as shown in the previous steps. 
```
